### ISSUE TITLE
GPS drains battery and heats phone up

### ISSUE THREAD

**Issue by Devun** [8890q/android_kernel_samsung_universal8890#5]
Hello,
I've noticed that using any app that makes use of the GPS causes the phone to heat up and drains the battery quite fast.

Could you please investigate this?

Big thanks for your awesome ROM!!!

Technical info:
Logs taken after having used Google Maps for a few minutes.
SM-G935F (S7 Edge Europe)
Android 19.1
[logcat_04-14-2022_10-14-26.txt](https://github.com/8890q/android_kernel_samsung_universal8890/files/8487520/logcat_04-14-2022_10-14-26.txt)
No tombstone
No kmsg
[dmesg.txt](https://github.com/8890q/android_kernel_samsung_universal8890/files/8487542/dmesg.txt)

If there is anything more you'd like to extract from my phone don't hesitate to tell me.

Big thanks in advance for your help!!
Devun / Azerty9971 (XDA)

**Comment by ivanmeler** [comment]
> Hello, I've noticed that using any app that makes use of the GPS causes the phone to heat up and drains the battery quite fast.
> 
> Could you please investigate this?
> 
> Big thanks for your awesome ROM!!!
> 
> Technical info: Logs taken after having used Google Maps for a few minutes. SM-G935F (S7 Edge Europe) Android 19.1 [logcat_04-14-2022_10-14-26.txt](https://github.com/8890q/android_kernel_samsung_universal8890/files/8487520/logcat_04-14-2022_10-14-26.txt) No tombstone No kmsg [dmesg.txt](https://github.com/8890q/android_kernel_samsung_universal8890/files/8487542/dmesg.txt)
> 
> If there is anything more you'd like to extract from my phone don't hesitate to tell me.
> 
> Big thanks in advance for your help!! Devun / Azerty9971 (XDA)

Oh hey i was testing gps for last 15 minutes on my G935F and it seems fine, phone didnt overheat and battery usage is fine (I had google maps open), What comes to mind is that your issue is coming from google apps since they also act as location provider proxy for apps that use gps and googles api for locations (Most of gps apps in this case) that would explain abnormal drain and heating up, Also please do not take logs with ```logcat -v long``` that produces unreadable logs with useless newlines instead use ```logcat -b all```.
My best advice in this case would be to try taking a full backup and reinstalling rom and gapps and see if this keep happening if not likely some cached file related to google location provider got stuck up and is causing drain

**Comment by Devun** [comment]
Hello,
I've redone the logcat with the new parameters and indeed, they are much more readable :).
[logcat_maps.txt](https://github.com/8890q/android_kernel_samsung_universal8890/files/8488960/logcat_maps.txt)

Indeed there seem to be an issue with package com.google.android.gms.location.history that is not found.
I actually reflashed my phone yesterday (using first the gapps you provided on the OP and then flashed the latest stable flamegapps).

I'll try first installing the maps addon to see if that registers it as location provider. Otherwise I'll change from Flamegapps to Nikgapps and retest :).

Big thanks in advance for your help!!
Azerty9971

**Comment by Devun** [comment]
Hello again,
Even after reinstalling with new GApps (used NikGApps full so that there would be no missing component) I still get:
`04-15 15:56:25.373  3772  4287 D libgps  : OnGpsExtensionMessage: message_id(1), data(0x6fd98c5d44), size(2792)
04-15 15:56:25.373  3772  4287 D libgps  : proxy__gnss_sv_status_cb: called. num_svs(10)
04-15 15:56:25.377  3772  4287 D libgps  : GpsiHook: API: gpsLocationCb(GpsiHookStateGps: e,c,G)
04-15 15:56:25.389  3886  3909 W AlarmManager: Unrecognized alarm listener com.android.server.location.gnss.GnssLocationProvider$$ExternalSyntheticLambda1@753b394
04-15 15:56:25.439 16454 16553 I RenderFPS: Avg render time in last 1000 millis: 8.520 (Actual: 11)
04-15 15:56:25.440  4666 16983 W GCoreUlr: Received locations too frequently: Location[fused 50.886262,4.470778 hAcc=11.2 et=+21h47m4s128ms alt=89.0 vAcc=1.3930944 vel=0.18621475 bear=257.43527 {Bundle[{indoorProbability=0.5348725, locationType=1}]}], Location[fused 50.886260,4.470690 hAcc=12.0 et=+21h47m14s108ms alt=89.0 vAcc=1.492899 vel=0.32488197 bear=251.10397 {Bundle[mParcelledData.dataSize=92]}]
04-15 15:56:25.440  4666 16983 I GCoreUlr: No valid location reports to store`

Is this a bug in the GApps, or could it be that the API to store the GPS location has changed?

EDIT: I've encountered this page: https://developer.android.com/reference/android/location/LocationProvider
Basically, this API is deprecated, could it be that libgps needs an update?

Thanks in advance for your help !!

Azerty9971
